app/services/agendaitem-and-subcase-properties-sync.js

```diff
@@ -0,0 +1,75 @@
+import Service, { inject as service } from '@ember/service';
+import CONSTANTS from 'frontend-kaleidos/config/constants';
+import moment from 'moment';
+import { setNotYetFormallyOk } from 'frontend-kaleidos/utils/agendaitem-utils';
+
+/**
+ * @description Set some properties on a model.
+ * @param model Kan van het type agendaitem of subcase zijn
+ * @param propertiesToSet de properties die we dienen aan te passen
+ * @param resetFormallyOk Dient de formaliteit aangepast te worden of niet (default true)
+ * @returns {Promise<*>}
+ */
+const setNewPropertiesToModel = async(model, propertiesToSet, resetFormallyOk = true) => {
+  if (resetFormallyOk) {
+    setNotYetFormallyOk(model);
+  }
+
+  const keys = Object.keys(propertiesToSet);
+  for (const key of keys) {
+    model.set(key, propertiesToSet[key]);
+  }
+
+  await model.save();
+  return model.reload();
```

Comment:
I didn't notice this before. Any idea why we do the `model.reload()` explicitely after the `model.save()` @ValenberghsSven ?

Reply:
not really sure, it's been in there for a long time.
Could be concurrency related
Could leave it in for now and use a seperate PR to see what jenkins does with it.

Some of those legacy utils are ... questionable to current code quality.
Lots of generic pieces of code to were used in shared components.

I wouldn't mind taking a stab at refactoring some of the utils to be more specific (save changes to subcase specifically instead of calling a method that also works for agendaitems)
In a follow-up ticket of course

Reply:
You're right. We can keep diving and improving :see_no_evil: Better to handle further improvements in another story (or in the margin of related stories)